proof/courier.go

```diff
@@ -21,18 +22,58 @@ import (
 
 // CourierType is an enum that represents the different types of proof courier
 // services.
-type CourierType int64
+//
+// TODO(ffranr): Rename to CourierProtocol.
+type CourierType string
 
 const (
 	// DisabledCourier is the default courier type that is used when no
 	// courier is specified.
-	DisabledCourier CourierType = iota
+	DisabledCourier CourierType = "disabled_courier"
 
 	// ApertureCourier is a courier that uses the hashmail protocol to
 	// deliver proofs.
-	ApertureCourier
+	//
+	// TODO(ffranr): Rename to HashmailCourier (use protocol name rather
+	//  than service).
+	ApertureCourier = "hashmail"
 )
 
+// NewCourierType returns the CourierType that corresponds to the given string.
+func NewCourierType(scheme string) (CourierType, error) {
+	switch scheme {
+	case ApertureCourier:
+		return ApertureCourier, nil
+	}
+
+	return DisabledCourier, fmt.Errorf("unknown courier address "+
+		"protocol: %v", scheme)
+}
+
+func ParseCourierAddr(addr string) (*url.URL, error) {
+	// Parse URI.
+	urlAddr, err := url.ParseRequestURI(addr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid proof courier URI address: %w",
+			err)
+	}
+
+	// Validate port number.
+	if urlAddr.Port() == "" {
```

Comment:
Port may not always be relevant. Instead I think `CourierType` could expose a method used for validating the courier based on its type. May want to spin it into an interface at that point as well. 

Non blocking for now though, but we'll want to remove this eventually to support more flexible courier types.

Reply:
Good point. Tracking here: https://github.com/lightninglabs/taproot-assets/issues/462